commit.go

```diff
@@ -107,21 +107,37 @@ type PushOptions struct {
 }
 
 var (
-	// commitPolicy bypasses any signing requirements when committing containers to images
-	commitPolicy = &signature.Policy{
-		Default: []signature.PolicyRequirement{signature.NewPRReject()},
-		Transports: map[string]signature.PolicyTransportScopes{
-			is.Transport.Name(): {
-				"": []signature.PolicyRequirement{
-					signature.NewPRInsecureAcceptAnything(),
-				},
-			},
+	// storageAllowedPolicyScopes overrides the policy for local storage
+	// to ensure that we can read images from it.
+	storageAllowedPolicyScopes = signature.PolicyTransportScopes{
+		"": []signature.PolicyRequirement{
+			signature.NewPRInsecureAcceptAnything(),
 		},
 	}
-	// pushPolicy bypasses any signing requirements when pushing (copying) images from local storage
-	pushPolicy = commitPolicy
 )
 
+// referenceOnlyImage wraps a reference in an UnparsedImage with methods that
+// always return no manifest and no signatures.  This is enough to be able to
+// speculatively evaluate "insecureAcceptAnything" and "reject" policies for
+// images which don't actually exist yet (i.e., references for which NewImage()
+// would fail).  Attempts to evaluate "signedBy" policies will fail because we
+// don't support retrieving signature information.
+type referenceOnlyImage struct {
+	dest types.ImageReference
+}
+
+func (r *referenceOnlyImage) Reference() types.ImageReference {
+	return r.dest
```

Comment:
(Strictly speaking, nothing quite promises not to call `types.UnparsedImage.Reference().NewImageSource()` and the like — but of course that would defeat the point of `UnparsedImage` fixing the contents of manifest+signatures in time against TOCTOU vulnerabilities.)

Reply:
Yeah, it's not a code path we currently go down, but it's a potential trouble spot in the future.